### PR TITLE
Update build target to java8, maven version to 3.5.2

### LINKED ISSUE #539: Move to java8...
Make appengine-plugins-core build and target Java 8.

I believe all our tools can now be moved?
@elharo ?
@etanshaul ?

### PR DESCRIPTION
**Pull request by loosebazooka**
fixes #539

This PR will mandate all further tooling builds target java8 or higher.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -258,7 +258,7 @@
           -->
           <instructions>
             <Bundle-SymbolicName>com.google.cloud.tools.appengine</Bundle-SymbolicName>
-            <Bundle-RequiredExecutionEnvironment>JavaSE-1.7</Bundle-RequiredExecutionEnvironment>
+            <Bundle-RequiredExecutionEnvironment>JavaSE-1.8</Bundle-RequiredExecutionEnvironment>
```
Comment (loosebazooka) [inline pom.xml]: @elharo @briandealwis I don't really know what any of this is, but I assume I need to do this?

File: .mvn/wrapper/maven-wrapper.properties [hunk 2]
```diff
@@ -1 +1 @@
-distributionUrl=https://storage.googleapis.com/cloud-tools-for-java-team-kokoro-build-cache/apache-maven-3.5.0-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.5.2/apache-maven-3.5.2-bin.zip
```
Comment (elharo) [inline .mvn/wrapper/maven-wrapper.properties]: we should probabkly put this in the google cache if we need it

Comment (loosebazooka) [reply .mvn/wrapper/maven-wrapper.properties]: Was this the cause of significant slowdowns previously? The history of this file starts off using the google cache version of this file, but we use the maven repo version in some other projects.

File: pom.xml (hunk 1 above)
Comment (briandealwis) [reply pom.xml]: :+1:

**Comment by loosebazooka** [comment]
Anyone want to put the github approval sticker on this? :)

**Pull request merged**